File: Python/Logic-1/answer_cell.py

# your cell phone rings. Return true if you should answer it
# normally you answer, except in the morning you only answer if it is your mom calling
# in all cases, if you are asleep, you do not answer
def answer_cell(is_morning, is_mom, is_asleep):
    b = True
    
    if is_asleep or (is_morning and not is_mom):
        b = False
    
    return b

print(answer_cell(False, False, False))
print(answer_cell(False, False, True))
print(answer_cell(True, False, False))

File: Java/Logic-1/answerCell.java

public class logic1 {

	// your cell phone rings. Return true if you should answer it
	// normally you answer, except in the morning you only answer if it is your mom calling
	// in all cases, if you are asleep, you do not answer
	public boolean answerCell(boolean isMorning, boolean isMom, boolean isAsleep) {
		
		boolean b = true;
		
		if (isAsleep || (isMorning && !isMom))
			b = false;
			
		return b;
		
	}
	
	public static void main(String[] args) {
		
		logic1 o = new logic1();
		
		System.out.println(o.answerCell(false, false, false));
		System.out.println(o.answerCell(false, false, true));
		System.out.println(o.answerCell(true, false, false));

	}

}
